### PR TITLE
[spirv] disable OpSource emission when DebugSource exists

### PR DESCRIPTION
**Pull request by jaebaek**

### CODE REVIEW

File: tools/clang/include/clang/SPIRV/SpirvModule.h [hunk 1]
```diff
@@ -127,6 +127,10 @@ class SpirvModule {
   // Adds the debug source to the module.
   void addSource(SpirvSource *);
 
+  // Clear sources if debugSources is not empty.
+  // void clearSourcesIfDebugSourceExist();
```
Comment (ehsannas) [inline tools/clang/include/clang/SPIRV/SpirvModule.h]: unused code

File: tools/clang/lib/SPIRV/EmitVisitor.cpp [hunk 2]
```diff
@@ -463,6 +462,8 @@ bool EmitVisitor::visit(SpirvString *inst) {
 }
 
 bool EmitVisitor::visit(SpirvSource *inst) {
+  // if (debugSourceExists) return true;
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/EmitVisitor.cpp]: unused code.

**Review by ehsannas — CHANGES_REQUESTED** [review]
Correct me if I'm wrong:
There's no need to add more member variables or member functions. EmitVisitor has access to SpirvOptions. You can just check if `-fspv-debug==rich`, and if so, then don't emit OpSource.

File: tools/clang/include/clang/SPIRV/SpirvModule.h (hunk 1 above)
Comment (jaebaek) [reply tools/clang/include/clang/SPIRV/SpirvModule.h]: removed

File: tools/clang/lib/SPIRV/EmitVisitor.cpp (hunk 2 above)
Comment (jaebaek) [reply tools/clang/lib/SPIRV/EmitVisitor.cpp]: removed

**Review by jaebaek — COMMENTED** [review]
I agree. I removed members and member functions. It uses `spvOptions.debugInfoRich` instead.

File: tools/clang/lib/SPIRV/EmitVisitor.cpp [hunk 3]
```diff
@@ -463,6 +461,9 @@ bool EmitVisitor::visit(SpirvString *inst) {
 }
 
 bool EmitVisitor::visit(SpirvSource *inst) {
+  if (spvOptions.debugInfoRich)
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/EmitVisitor.cpp:467]: Please add a comment here:
```cpp
// We should either emit OpSource or DebugSource, not both.
// Therefore if rich debug info is being generated, we will skip emitting OpSource.
```

Comment (jaebaek) [reply tools/clang/lib/SPIRV/EmitVisitor.cpp:467]: done

**Review by ehsannas — APPROVED** [review]
Thanks

**Pull request merged**